File: sdks/c++/basys.sdk.cc/src/submodel/submodel/connected/submodelelement/property/ConnectedReferenceElement.cpp
/*
 * ConnectedReferenceElement.cpp
 *
 *      Author: wendel
 */

#include "ConnectedReferenceElement.h"

namespace basyx {
namespace submodel {

ConnectedReferenceElement::ConnectedReferenceElement(std::shared_ptr<vab::core::proxy::IVABElementProxy> proxy) :
  ConnectedDataElement(proxy)
{}

void ConnectedReferenceElement::setValue(const std::shared_ptr<IReference> & ref)
{
  //todo
  //this->setProxyValue(submodelelement::property::PropertyPaths::VALUE, ref);
}

std::shared_ptr<IReference> ConnectedReferenceElement::getValue() const
{
  //todo 
  return nullptr;// this->getProxyValue(submodelelement::property::PropertyPaths::VALUE);
}

}
}

File: sdks/java/basys.sdk/src/main/java/org/eclipse/basyx/submodel/metamodel/connected/submodelelement/ConnectedReferenceElement.java
package org.eclipse.basyx.submodel.metamodel.connected.submodelelement;

import java.util.Map;

import org.eclipse.basyx.submodel.metamodel.api.reference.IReference;
import org.eclipse.basyx.submodel.metamodel.api.submodelelement.IReferenceElement;
import org.eclipse.basyx.submodel.metamodel.map.reference.Reference;
import org.eclipse.basyx.submodel.metamodel.map.submodelelement.property.Property;
import org.eclipse.basyx.vab.modelprovider.VABElementProxy;

/**
 * "Connected" implementation of IReferenceElement
 * @author rajashek
 *
 */
public class ConnectedReferenceElement extends ConnectedDataElement implements IReferenceElement {
	public ConnectedReferenceElement(VABElementProxy proxy) {
		super(proxy);		
	}
	

	@Override
	public void setValue(IReference ref) {
		getProxy().setModelPropertyValue(Property.VALUE, ref);
		
	}

	@SuppressWarnings("unchecked")
	@Override
	public IReference getValue() {
		return Reference.createAsFacade((Map<String, Object>) getElem().getPath(Property.VALUE));
	}

}